ocaml/runtime/gc_stats.c

```diff
@@ -57,11 +57,18 @@ void caml_accum_alloc_stats(
   acc->forced_major_collections += s->forced_major_collections;
 }
 
+double caml_minor_words(caml_domain_state *local)
+{
+  return local->stat_minor_words
+    + (double) Wsize_bsize ((uintnat) local->young_end -
+                            (uintnat) local->young_ptr);
+}
+
 void caml_collect_alloc_stats_sample(
   caml_domain_state *local,
   struct alloc_stats *sample)
 {
-  sample->minor_words = local->stat_minor_words;
+  sample->minor_words = caml_minor_words(local);
```

Comment:
I think this change is in the wrong place, since this only makes sense to do for the current domain (see other comment)